packages/mobile/src/invite/EnterInviteCode.tsx

```diff
@@ -94,30 +94,39 @@ export class EnterInviteCode extends React.Component<Props, State> {
   state: State = {
     isSubmitting: false,
     appState: AppState.currentState,
-    validCodeInClipboard: false,
+    validCode: null,
   }
 
   async componentDidMount() {
-    AppState.addEventListener('change', this.handleValidCodeInClipboard)
+    AppState.addEventListener('change', this.handleAppStateChange)
+    this.checkForReferrerCode()
     this.checkIfValidCodeInClipboard()
   }
 
   componentWillUnmount() {
-    AppState.removeEventListener('change', this.handleValidCodeInClipboard)
+    AppState.removeEventListener('change', this.handleAppStateChange)
   }
 
   openMessage = () => {
     SendIntentAndroid.openSMSApp()
   }
 
+  checkForReferrerCode = async () => {
+    const validCode = await getInviteCodeFromReferrerData()
+    if (validCode) {
+      this.setState({ validCode })
+    }
+  }
+
   checkIfValidCodeInClipboard = async () => {
     const message = await Clipboard.getString()
     const validCode = extractValidInviteCode(message)
-
-    this.setState({ validCodeInClipboard: validCode !== null })
+    if (validCode) {
+      this.setState({ validCode })
```

Comment:
please run this by Corey who intentionally wanted the user to click.
I'm fine with this approach.